### DISCUSSION TITLE
Inter module dependency detection

### DISCUSSION THREAD

**Discussion by hengsin** [JaneWardSandy/eclipse-pde-partial-idea#70] [Q&A] [answered]
Hi, will the plugin auto add dependency between modules through required-bundle and import-package inspection ? It seems I've to manually add inter module dependency to resolve build error.

Thanks.

**Comment by JaneWardSandy** [comment]
You can request the PDE plugins resolve the dependencies list and order by using the following step:

 - `right click` editor in manifest.mf file, and click `resolve manifest`. 
 It will re-read the manifest.mf, and remove all `modules dependency` and all dependencies that start with `Partial: `.
Then add `Bundle-ClassPath` dependency, modules dependency, and project dependencies.
And re-order the dependency entry by manifest.mf
<img width="545" alt="image" src="https://user-images.githubusercontent.com/15345355/154924819-11d39a55-a107-466a-a802-29e3649cae5f.png">

- `right click` any node in project navigator,  and click `resolve workspace`
It will re-read the `target definition`, then resolve all module.
<img width="464" alt="image" src="https://user-images.githubusercontent.com/15345355/154925262-443c95f0-216e-433b-8e53-6f73a5075474.png">

May it can help

**Reply by hengsin** [reply]
Not working for me. Will test later whether I can reproduce that with a minimal example project.

IntelliJ IDEA 2021.3.2 (Community Edition)
Build #IC-213.6777.52, built on January 28, 2022

**Comment by hengsin** [comment]
[test.zip](https://github.com/JaneWardSandy/eclipse-pde-partial-idea/files/8109673/test.zip)

Hi, I've attached a minimal, dummy project to illustrate the issue. Neither "Resolve Workspace" or "Resolve Manifest" can help to resolve the build error of module2 (package p1 missing).

**Reply by JaneWardSandy** [reply]
I had duplicate this issue, I will fix it in next version

**Comment by JaneWardSandy** [comment, marked as answer]
This bug has been fix and release at v1.3.6.1

**Reply by hengsin** [reply]
Install the update from market place ?

**Reply by JaneWardSandy** [reply]
You can update from market place when IntelliJ approve new public tomorrow, or you can download from [release of Github](https://github.com/JaneWardSandy/eclipse-pde-partial-idea/releases/tag/v1.3.6.1) then install manually if you are in a hurry.

**Reply by hengsin** [reply]
I've install from release and it is working great now, thanks!
